sdk/inc/azure/core/az_span.h

```diff
@@ -303,6 +303,7 @@ az_span az_span_copy_u8(az_span destination, uint8_t byte);
  */
 AZ_INLINE void az_span_fill(az_span destination, uint8_t value)
 {
+  // NOLINTNEXTLINE(clang-analyzer-security.insecureAPI.DeprecatedOrUnsafeBufferHandling)
```

Comment:
The deprecation is not about memset vs memcpy, it is memset vs memset_s and memcpy vs memcpy_s. We can't use the `_s` variants because those are not available on C99.

Here's the full warning we are suppressing:
> warning: Call to function 'memset' is insecure as it does not provide security checks introduced in the C11 standard. Replace with analogous functions that support length arguments or provides boundary checks such as 'memset_s' in case of C11 

https://en.cppreference.com/w/c/string/byte/memset

Reply:
yes, so, we can leave the warning disable for memmove and replace this memset for a loop routine that calls az_copy to set each destination.

Same thing at the end.. just one less warning to be ignored

Reply:
I prefer to leave things as is. I am not sure that the perf characteristics are identical. Feel free to try it out and submit a fix in a separate PR.